index.js

```diff
@@ -2629,20 +2639,21 @@ RemoteObjectTemplate.bindDecorators = function (objectTemplate) {
 
 	    // Called first time with parameter rather than target - call supertypes supertypeClass function which will
         // return a function that must be called on the 2nd pass when we have a target.  It will remember parameter
-        ret = ObjectTemplate.supertypeClass(target, objectTemplate);
+        ret = ObjectTemplate2.supertypeClass(target, objectTemplate);
```

Comment:
should this be `ObjectTemplate` or `ObjectTemplate2` ?

Reply:
Hmm, I should rename this. `ObjectTemplate2` is the entire "SupertypeModule", the whole shebang. Part of the work for typescripting Supertype is splitting it out into different components, so now we have an `ObjectTemplate` class, which is basically the `amorphicStatic` reference. 

In this snippet of code I'm storing the module of supertype (e.g. `require('supertype')` in a variable called `ObjectTemplate2`. If I want to access the exported `supertypeClass` property from the supertype module, I can access it like `ObjectTemplate2.supertypeClass`. This is similar to just 
```
var _ = require('underscore');
_.find(...)
```
or whatever.

`ObjectTemplate` in this snippet of code refers to the definition of the _root amorphic object itself_. This is an actual object, a real construct.

I will rename `ObjectTemplate2` to `SupertypeModule` to be more explicit about what it is. I meant to do so beforehand but completely forgot. Nice catch.